src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,236 @@
+import { toBNWei } from "../utils";
+import { MockUBAConfig } from "../clients/mocks";
+import { computePiecewiseLinearFunction } from "./UBAFeeUtility";
+
+describe("UBAFeeSpokeCalculatorAnalog", () => {
+  describe.only("getEventFee", () => {
+    // Create a chain Id that we can reference anywhere
+    // without any magic numbers.
+    const chainId = "1";
+
+    // This is the base case for this function. We want to ensure that if we have a
+    // positive balancing fee result, that the result is equal to the integration of
+    // the balancing fee curve without any additional modifiers.
+    it("should expect if our balancing fee is positive, that the result is a single integration", () => {
+      // We start with a basic config that's been instantiated.
+      const config = new MockUBAConfig();
+      // We set the balancing fee curve to be a basic curve that has a positive slope.
+      // We can work in terms of any decimal as long as we're consistent & the second value
+      // in the tuple is in terms of wei or any other 18 decimal token.
+      config.setBalancingFeeCurve(chainId, [
+        [toBNWei(0, 6), toBNWei(0, 0)],
+        [toBNWei(1, 6), toBNWei(0.2, 18)],
+      ]);
+      // We set the reward multiplier to be 1. This is to ensure that we don't have any
+      // additional reward multiplier that would affect our calculations.
+      config.setRewardMultiplier(chainId, utils.parseEther("1"));
+      // We set the amount, lastRunningBalance, and lastIncentiveBalance for our test.
+      // Note: we need to ensure that this matches the fixed decimals of the balancing fee curve.
+      const amount = toBNWei(10, 6);
+      const lastRunningBalance = toBNWei(1000, 6);
+      const lastIncentiveBalance = toBNWei(1000, 6);
+      // We call the getEventFee function with the parameters we've set above.
+      const fee = getEventFee(amount, "inflow", lastRunningBalance, lastIncentiveBalance, 1, config).balancingFee;
+      // We are expecting that the fee will be positive. As a result, let's assert
+      // that the fee is greater than 0.
+      expect(fee.gt(0)).toBeTruthy();
+      // Because the fee is greater than zero, we applied no additional reward multiplier,
+      // or any other factors that would affect the fee, we expect that the fee will be
+      // equal to the integration of the balancing fee curve.
+      const integration = computePiecewiseLinearFunction(
+        config.getBalancingFeeTuples(Number(chainId)),
+        lastRunningBalance,
+        lastRunningBalance.add(amount)
+      );
+      // We expect that the fee is equal to the integration of the balancing fee curve.
+      // Note: we need to ensure that we're using the same decimals as the balancing fee curve.
+      // Note: The expectation above is that our fee is positive
+      expect(fee.toString()).toEqual(integration.toString());
+
+      // As an edge case, we can also test that if the balancing fee that is initially computed
```

Comment:
Should we make this edge case a separate unit test in its own `it(...)`?

Reply:
I personally think we should keep it as is. The reason being: if we separated it we'd have a lot of code duplication